audit-log-api/src/use-cases/RetryMessageUseCase.ts

```diff
@@ -27,23 +29,38 @@ export default class RetryMessageUseCase {
       return eventValidationResult
     }
 
-    const eventXmlContent = await this.retrieveEventXmlFromS3UseCase.retrieve(event.s3Path)
+    const eventContent = await this.retrieveEventXmlFromS3UseCase.retrieve(event.s3Path)
+
+    if (isError(eventContent)) {
+      return eventContent
+    }
+
+    let eventJsonContent
```

Comment:
FWIW this verbose data validation could avoided by using this [validation library](https://github.com/colinhacks/zod), obvs no expectation to change this to use the library right now but I find it's a really useful library to know about.

Reply:
I haven't seen `zod` before - it looks like a handy library. Good shout! I think we should definitely look at integrating that and being a bit more strict about our JSON parsing - there are probably quite a few areas within audit logging that could make use of that.